presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaMetadata.java

```diff
@@ -48,19 +58,30 @@
 public class KafkaMetadata
         implements ConnectorMetadata
 {
+    private static final String AUTO_RESOLVE = "auto";
+    private static final String AVRO = "avro";
+
     private final boolean hideInternalColumns;
     private final Map<SchemaTableName, KafkaTopicDescription> tableDescriptions;
+    private final Optional<SchemaRegistryClient> schemaRegistryClient;
+    private final AvroSchemaConverter avroSchemaConverter;
```

Comment:
Schema discovery is becoming quite complicated logic now. I would extract this out of `KafkaMetadata` and made it clear when the schema is read from where.

Also, is it possible to not declare tables at all in `tableDescriptions` so kafka connector could discovers topics automatically? Thanks to that it would be much easier to configure kafka connector.

Reply:
Yes it is! We have code that reads schema from schema-registry without json files, will add that as well.